test/e2e-go/features/devmode/devmode_test.go

```diff
@@ -54,14 +53,15 @@ func TestDevMode(t *testing.T) {
 
 	// 2 transactions should be sent within one normal confirmation time.
 	for i := uint64(0); i < 2; i++ {
-		txn = fixture.SendMoneyAndWait(firstRound+i, 100000, 1000, sender.Address, receiver.String(), "")
-		require.Equal(t, firstRound+i, txn.Txn.Txn.FirstValid)
+		txn = fixture.SendMoneyAndWait(firstRound+i, 100001, 1000, sender.Address, receiver.String(), "")
+		// SendMoneyAndWait subtracts 1 from firstValid
+		require.Equal(t, firstRound+i-1, uint64(txn.Txn.Txn.FirstValid))
 	}
 	require.True(t, time.Since(start) < 8*time.Second, "Transactions should be quickly confirmed faster than usual.")
 
 	// Without transactions there should be no rounds even after a normal confirmation time.
 	time.Sleep(10 * time.Second)
 	status, err := fixture.LibGoalClient.Status()
 	require.NoError(t, err)
-	require.Equal(t, txn.ConfirmedRound, status.LastRound, "There should be no rounds without a transaction.")
+	require.Equal(t, *txn.ConfirmedRound, status.LastRound, "There should be no rounds without a transaction.")
```

Comment:
How did this ever pass? The following simple test fails:

```go
type PointerValue struct {
	Pointer *int
}

func TestPointerValue(t *testing.T) {
	zero := 0
	pv := PointerValue{&zero}
	require.Equal(t, 0, pv.Pointer)
}
```

Reply:
(this is an argument for generics)

Reply:
I'm not sure how it passed before...I suspect it wasn't returning a pointer at some point?